packages/pyright-internal/src/common/uri/memoization.ts

```diff
@@ -55,7 +55,7 @@ export function cacheStaticFunc() {
     return function cacheStaticFunc_Fast(target: any, functionName: string, descriptor: PropertyDescriptor) {
         const originalMethod = descriptor.value;
         descriptor.value = function (...args: any) {
-            const key = `${functionName}+${args.map((a: any) => a?.toString()).join(',')}`;
+            const key = `${functionName}+${args?.map((a: any) => a?.toString()).join(',')}`;
```

Comment:
If there are no args, this will presumably result in a key that has `undefined` in it. Won't that same key result from a function that is passed a single argument of type `undefined`? Just want to make sure this isn't a cache namespace collision.

Reply:
I don't believe this is possible if the function was defined with arguments, args will always be defined in that case. So for a function `foo`, the argument part of the key is undefined when:
- There are no arguments ever
- There's one argument and it's undefined

But you can't have the same situation for the same function because it either has an argument or it doesn't (in the definition of the function). I believe that means there should be no cache collision.

Reply:
I tried this with some test code just to make sure:

```typescript
function cacheStaticFunc() {
    return function cacheStaticFunc_Fast(target: any, functionName: string, descriptor: PropertyDescriptor) {
        const originalMethod = descriptor.value;
        descriptor.value = function (...args: any) {
            const key = `${functionName}+${args?.map((a: any) => a?.toString()).join(',')}`;
            console.log(`Calling ${functionName} with args: ${key}`);
            return originalMethod.apply(this, args);
        };
        return descriptor;
    };
}

class Test {

  @cacheStaticFunc()
  static foo() {
   console.log('foo');
}
  @cacheStaticFunc()
  static bar(arg1?: string, arg2?: string) {
   console.log('bar');
}
}

Test.foo();
Test.bar();
Test.bar(undefined, undefined);
Test.bar('foo');
```

Produces the following output:
```
Calling foo with args: foo+
foo
Calling bar with args: bar+
bar
Calling bar with args: bar+,
bar
Calling bar with args: bar+foo
bar
```

Reply:
Which I believe is following what I thought. If you pass undefined to a function with a single argument, it's the same as not passing an argument. Which internally (in the function) makes sense. That argument would be undefined if you passed nothing.